source/Nevermore/IRelationalStoreConfiguration.cs

```diff
@@ -54,15 +55,27 @@ public interface IRelationalStoreConfiguration
 
         /// <summary>
         /// Gets or sets whether Multiple Active Result Sets is enabled. On .NET Core 3.1 this has issues on Linux, so it
-        /// defaults to false. 
+        /// defaults to false.
         /// </summary>
         /// <remarks>https://docs.microsoft.com/en-us/sql/relational-databases/native-client/features/using-multiple-active-result-sets-mars?view=sql-server-ver15</remarks>
         bool ForceMultipleActiveResultSets { get; set; }
-        
+
         /// <summary>
         /// Gets or sets whether to actively detect similar queries being executed in a way that is slightly different,
         /// resulting in duplicate query plans being created.
         /// </summary>
         public bool DetectQueryPlanThrashing { get; set; }
+
+        /// <summary>
+        /// Defines a name generator to be used when creating global temp tables during related document updates. <para />
+        /// This is used when there are more related documents than can fit in a single SQL command
+        /// </summary>
+        Func<string> RelatedDocumentsGlobalTempTableNameGenerator { get; }
+
+        /// <summary>
+        /// Used with <see cref="RelatedDocumentsGlobalTempTableNameGenerator"/> when transferring data from the temp table to the related documents table to ensure collation matches. <para />
+        /// Defaults to SQL_Latin1_General_CP1_CS_AS
+        /// </summary>
+        public string RelatedDocumentsDatabaseCollation { get; set; }
```

Comment:
See previous discussion: https://github.com/OctopusDeploy/Nevermore/pull/153#discussion_r659399751

> 
> 
> So it looks like the column collation in the temporary table needs to match the collation from the related document table column when performing the join. The reason why `SQL_Latin1_General_CP1_CS_AS` is there is because that's how we set up the test database.
> 
> We would either need to:
> 
>     1. query the database schema to determine what the collation of the related document table columns are, or
> 
>     2. find a way around not using global temp tables

Reply:
I'm not convinced static configuration is suitable for this - it's not great experience for the caller to set this.

What if we injected a func that lazily queried the collation from the database?

We do something similar with the key allocator already (make DB calls during query text generation)
https://github.com/OctopusDeploy/Nevermore/blob/516d046eb4f427c442ccd7c4b6d7e15bc597cd98/source/Nevermore/Util/DataModificationQueryBuilder.cs#L33

Reply:
What about populating `@reference` table multiple times, once per command?

Reply:
> query the database schema to determine what the collation of the related document table columns are, or

I don't think Nevermore make this type of queries anywhere so existing sql users might not have enough permissions.

Reply:
> 
> 
> What about populating `@reference` table multiple times, once per command?

Not enough parameters in the command available to do this.

Reply:
what id we create a real table in the database?

Reply:
We can query for the collation information instead of having it as a configuration value, but this requires `VIEW SERVER STATE` or `VIEW DATABASE STATE` permissions which doesn't seem unreasonable.

Reply:
I was able to determine the collation dynamically using the above SQL. 
It now works regardless of the database column collations as long as the executor has the permission mentioned above.

https://github.com/OctopusDeploy/Nevermore/pull/153/files#diff-94be40892362f618a2307a0d89bf656db3bcf5ab10f60619ac87954a1800647aR492